### PR TITLE
[CBRD-26227] Add sql test case for preventing non-group-by columns from becoming NULL in WITH ROLLUP when only_full_group_by is disabled

### PR DESCRIPTION
**Pull request by kimhak-tech**
Refer to http://jira.cubrid.org/browse/CBRD-26227 

**CBRD-26227** fixes an issue in which columns not specified in the GROUP BY clause were incorrectly output as `NULL` in the `WITH ROLLUP` summary row when `only_full_group_by = NO`.

The expected behavior is that non-group-by columns should preserve their evaluated values unless the actual data is `NULL`.

### CODE REVIEW

**Review by junsklee — APPROVED** [review]

**Review by ssihil — APPROVED** [review]

**Review by kwonhoil — APPROVED** [review]

**Review by tw-kang — APPROVED** [review]

**Review by Hamkua — APPROVED** [review]

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+INSERT INTO t1 VALUES
+(1, 1, 'a'),
+(2, 2, 'b'), (2, 2, 'b'),
+(3, 3, 'c'), (3, 3, 'c'), (3, 3, 'c');
+
+set system parameters 'only_full_group_by=no';
+
+evaluate 'Case 1: Single non-grouped column(c1) with ROLLUP';
+select /*+ recompile */ c1, c2 from t1 group by c2 with rollup;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2 from t1 group by c2 with rollup order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+evaluate 'Case 2: Multiple non-grouped columns(c1, c3) with ROLLUP';
+select /*+ recompile */ c1, c2, c3 from t1 group by c2 with rollup;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2, c3 from t1 group by c2 with rollup order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+evaluate 'Case 3: Non-grouped column with Aggregate function';
+select /*+ recompile */ c1, c2, count(*) as cnt from t1 group by c2 with rollup;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2, count(*) as cnt from t1 group by c2 with rollup order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+evaluate 'Case 4: Non-grouped column with two GROUP BY items';
+select /*+ recompile */ c1, c2, c3 from t1 group by c2, c3 with rollup;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2, c3 from t1 group by c2, c3 with rollup order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+);
+
+INSERT INTO t2 VALUES
+(1, 1, 'a'),
+(NULL, 2, 'b'),
+(3, NULL, 'c'),
+(NULL, NULL, NULL);
+
+evaluate 'Case 5: Non-grouped column contains actual NULL values';
+select /*+ recompile */ c1, c2 from t2 group by c2 with rollup;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2 from t2 group by c2 with rollup order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+evaluate 'Case 6: Non-grouped column with ORDER BY';
+select /*+ recompile */ c1, c2 from t2 group by c2 with rollup order by c2;
+
+evaluate 'Case 7: Non-grouped column with HAVING and Aggregate condition';
+select /*+ recompile */ c1, c2, count(*) as cnt from t2 group by c2 with rollup having count(*) >= 1;
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
select /*+ recompile */ c1, c2, count(*) as cnt from t2 group by c2 with rollup having count(*) >= 1 order by c2;
```

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+set system parameters 'only_full_group_by=no';
+
+evaluate 'Case 1: Single non-grouped column(c1) with ROLLUP';
+select /*+ recompile */ c1, c2 from t1 group by c2 with rollup;
+
+evaluate 'Case 2: Multiple non-grouped columns(c1, c3) with ROLLUP';
+select /*+ recompile */ c1, c2, c3 from t1 group by c2 with rollup;
+
+evaluate 'Case 3: Non-grouped column with Aggregate function';
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]:
```suggestion
evaluate 'Case 3: Non-grouped column + aggregate column mixed with ROLLUP';
```

Comment (bagus-kim) [reply sql/_13_issues/_25_2h/cases/cbrd_26227.sql]: 비그룹 컬럼+집계 컬럼 혼합인 케이스 같습니다.그러므로 주석 수정이 필요할 것 같습니다.(제가 제안한것은 참고용으로 사용해주세요)

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql (hunk 6 above)
Comment (bagus-kim) [reply sql/_13_issues/_25_2h/cases/cbrd_26227.sql]: having count(*) >=1 은 항상 '참'임으로 변별력이 낮을 것 같습니다. >=2등으로 변경해보는 것은 어떠할까요?

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    c1 int,
+    c2 float,
+    c3 varchar(20)
+);
+
+INSERT INTO t2 VALUES
+(1, 1, 'a'),
+(NULL, 2, 'b'),
+(3, NULL, 'c'),
+(NULL, NULL, NULL);
```
Comment (bagus-kim) [inline sql/_13_issues/_25_2h/cases/cbrd_26227.sql]: t2에서 c2=NULL이 2개 존재합니다. 변별력을 높이기 위해 아래와 같이 변경해보는 것은 어떠할까요?

INSERT INTO t2 VALUES
(1, 1, 'a'),
(NULL, 2, 'b'),
(3, 3, 'c'),
(NULL, 4, NULL);

case6 : Order by c2,c1으로 변경하면 더 좋겠습니다.

File: sql/_13_issues/_25_2h/cases/cbrd_26227.sql (hunk 4 above)
Comment (bagus-kim) [reply sql/_13_issues/_25_2h/cases/cbrd_26227.sql]: 답안지를 위해 order by를 넣어야 하는데 group by도 봐야 한다면, 추가case로 order by를 하면 될듯합니다

**Comment by bagus-kim** [comment]
추가 검토 시나리오 입니다.

--c1+1, upper(c3) 표현식 사용
evaluate 'Case9: Non-grouped expression columns with ROLLUP';
select /*+ recompile */ c1 + 1 as c1p, c2, upper(c3) as c3u
from t1
group by c2 with rollup
order by c2;

--에러가 일관되게 발생하는지 
evaluate 'Case10: only_full_group_by=YES should error for multiple non-grouped columns too';
set system parameters 'only_full_group_by=yes';
select /*+ recompile */ c1, c2, c3 from t1 group by c2 with rollup;
set system parameters 'only_full_group_by=no';

-- groupby의 NULL과 rollup NULL구분 확인
DROP TABLE IF EXISTS t3;
CREATE TABLE t3 (c1 int, c2 float, c3 varchar(20));

-- c2 NULL 그룹을 명확히 만들고, 총계 행(ROLLUP NULL)과 구분되게 유도
-- NULL 그룹의 대표값을 999
INSERT INTO t3 VALUES
(1, 1, 'a'),
(2, 2, 'b'),
(3, 3, 'c'),
(999, NULL, 'null_group'),
(999, NULL, 'null_group');

evaluate 'Case11: GROUP BY key contains NULL (must produce two c2=NULL rows distinguishable by c1)';
select /*+ recompile */ c1, c2
from t3
group by c2 with rollup
order by c2, c1;

DROP TABLE IF EXISTS t3;

**Comment by kimhak-tech** [comment]
> 추가 검토 시나리오 입니다.
> 
> --c1+1, upper(c3) 표현식 사용 evaluate 'Case9: Non-grouped expression columns with ROLLUP'; select /*+ recompile */ c1 + 1 as c1p, c2, upper(c3) as c3u from t1 group by c2 with rollup order by c2;
> 
> --에러가 일관되게 발생하는지 evaluate 'Case10: only_full_group_by=YES should error for multiple non-grouped columns too'; set system parameters 'only_full_group_by=yes'; select /*+ recompile */ c1, c2, c3 from t1 group by c2 with rollup; set system parameters 'only_full_group_by=no';
> 
> -- groupby의 NULL과 rollup NULL구분 확인 DROP TABLE IF EXISTS t3; CREATE TABLE t3 (c1 int, c2 float, c3 varchar(20));
> 
> -- c2 NULL 그룹을 명확히 만들고, 총계 행(ROLLUP NULL)과 구분되게 유도 -- NULL 그룹의 대표값을 999 INSERT INTO t3 VALUES (1, 1, 'a'), (2, 2, 'b'), (3, 3, 'c'), (999, NULL, 'null_group'), (999, NULL, 'null_group');
> 
> evaluate 'Case11: GROUP BY key contains NULL (must produce two c2=NULL rows distinguishable by c1)'; select /*+ recompile */ c1, c2 from t3 group by c2 with rollup order by c2, c1;
> 
> DROP TABLE IF EXISTS t3;

I have reflected all the comments. Thank you for your suggestion.

**Review by bagus-kim — APPROVED** [review]
